library/src/main/java/com/getstream/sdk/chat/adapter/ChannelListDiffCallback.java

```diff
@@ -54,21 +55,14 @@ public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
             return false;
         }
         // Check Message Update
-        try {
-            if (oldChannel.getChannelState().getLastMessage().getUpdatedAt().getTime() < newChannel.getChannelState().getLastMessage().getUpdatedAt().getTime()) {
-                return false;
-            }
-        }catch (Exception e){
-            e.printStackTrace();
+        Message oldLastMessage = oldChannel.getChannelState().getLastMessage();
+        Message newLastMessage = newChannel.getChannelState().getLastMessage();
+        if (oldLastMessage != null && oldLastMessage.getUpdatedAt().getTime() < newLastMessage.getUpdatedAt().getTime()) {
+            return false;
         }
-
         // Check Message Delete
-        try {
-            if (oldChannel.getChannelState().getLastMessage() != newChannel.getChannelState().getLastMessage()) {
-                return false;
-            }
-        }catch (Exception e){
-            e.printStackTrace();
+        if (oldLastMessage != null && oldLastMessage != newLastMessage) {
```

Comment:
You're comparing references, but actual messages I assume should be compared here:
```java
oldLastMessage.equals(newLastMessage)
```
And by the way, there is java utility class which compares objects and check nulls:
```java
java.util.Objects.equals(oldLastMessage, newLastMessage)
```

Reply:
yeah, those are Objects. :)